alicloud/resource_alicloud_ddosbgp_instance.go

```diff
@@ -0,0 +1,212 @@
+package alicloud
+
+import (
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/bssopenapi"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ddosbgp"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudDdosbgpInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudDdosbgpInstanceCreate,
+		Read:   resourceAlicloudDdosbgpInstanceRead,
+		Update: resourceAlicloudDdosbgpInstanceUpdate,
+		Delete: resourceAlicloudDdosbgpInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Required:     false,
+				Default:      "1",
+				ForceNew:     true,
+				ValidateFunc: validateDdosbgpInstanceName,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Required:     false,
+				ValidateFunc: validateDdosbgpInstanceName,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"base_bandwidth": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bandwidth": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_count": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"period": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validateAllowedIntValue([]int{1, 2, 3}),
+				Optional:     true,
+				Default:      1,
+				ForceNew:     true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudDdosbgpInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := buildDdosbgpCreateRequest(d, meta)
+
+	raw, err := client.WithBssopenapiClient(func(bssopenapiClient *bssopenapi.Client) (interface{}, error) {
+		return bssopenapiClient.CreateInstance(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_ddosbgp_instance", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+	resp := raw.(*bssopenapi.CreateInstanceResponse)
+	// execute errors including in the bssopenapi response
+	if !resp.Success {
```

Comment:
I think the line 90 is redundant. If the err is not nil, the response is success and there is no need to check again.